### ISSUE TITLE
[Bug]: MSBuild packages 17.3.4 not published on nuget.org

### ISSUE THREAD

**Issue by tmat** [dotnet/msbuild#10492]
### Issue Description

Packages should be published

### Steps to Reproduce

Packages should be published

### Expected Behavior

Packages should be published

### Actual Behavior

Packages not published

**Comment by baronfel** [comment]
Those are pretty old - where is this dependency version coming from?

**Comment by tmat** [comment]
That's the version that source build requires.

**Comment by tmat** [comment]
See https://github.com/dotnet/roslyn/blob/main/eng/Directory.Packages.props#L28-L53

**Comment by maridematte** [comment]
It seems that this was a change in main from sometime ago: https://github.com/dotnet/roslyn/pull/73283. The problem of the package not being published was discussed on the PR. Has anything else changed recently that could cause this issue?

**Comment by ViktorHofer** [comment]
> That's the version that source build requires.

You can use a newer version of MSBuild dependencies. I.e. 17.8.3 is available on SBRP: https://github.com/dotnet/source-build-reference-packages/tree/main/src/referencePackages/src/microsoft.build

**Comment by ViktorHofer** [comment]
cc @dotnet/source-build-internal

**Comment by ellahathaway** [comment]
> > That's the version that source build requires.
> 
> Source-build doesn't require that old version of MSBuild. Any package version that exists in SBRP is fine. You can use a newer version of MSBuild dependencies. I.e. 17.8.3 is available on SBRP: https://github.com/dotnet/source-build-reference-packages/tree/main/src/referencePackages/src/microsoft.build

This is correct. I would not expect this package to be published on nuget.org because it is older. The package should be being pulled from SBRP.
